src/WooCommerce.php

```diff
@@ -721,4 +721,26 @@ public static function getProductAttributes(\WC_Product $product) {
     return $data;
   }
 
+  /**
+   * Add salutation field for billing and shipping.
+   *
+   * @implements woocommerce_checkout_fields
+   */
+  public static function woocommerce_checkout_fields(?array $fields): array {
+    $fields['shipping']['shipping_salutation'] = [
+      'label' => __('Salutation', Plugin::L10N),
+      'class' => ['form-row-wide'],
+      'type' => 'select',
+      'options' => [
+        __('Mr', Plugin::L10N) => __('Mr', Plugin::L10N),
+        __('Ms', Plugin::L10N) => __('Ms', Plugin::L10N),
+        __('Company', Plugin::L10N) => __('Company', Plugin::L10N),
```

Comment:
If we are implementing this as a standard in 2019, then I think there should be at least an option "Diverse" or alternatively "Prefer not to say" here.

Reply:
We discussed this and decided against for now.

Reply:
Forgot to mention, this is an optional field, which brings me to the point of adding a null value.